src/main/java/org/spongepowered/api/map/MapCanvas.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.map;
+
+import com.flowpowered.math.vector.Vector2i;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.data.persistence.DataBuilder;
+import org.spongepowered.api.data.persistence.InvalidDataException;
+import org.spongepowered.api.map.color.MapColor;
+import org.spongepowered.api.map.color.MapColorTypes;
+
+import java.awt.Image;
+import java.util.Optional;
+
+/**
+ * Represents the canvas of a {@link org.spongepowered.api.item.inventory.ItemStack}
+ * with {@link org.spongepowered.api.item.ItemType} {@link org.spongepowered.api.item.ItemTypes#FILLED_MAP}
+ * Changes to maps will not apply until the client relogs
+ */
+public interface MapCanvas extends DataSerializable {
+
+    static Builder builder() {
+        return Sponge.getRegistry().createBuilder(Builder.class);
+    }
+
+    /**
+     * Gets the map color at a location on the canvas
+     * @param x int x location
+     * @param y int y location
+     * @return MapColor color at this location
```

Comment:
You don't need to repeat the argument name or the return type, it's already implied.. You just comment on what the parameter is doing or describing it itself, or the return value, whether it's a copy, mutable, live, editable etc.